aptos-move/block-executor/src/executor.rs

```diff
@@ -478,13 +509,37 @@ where
             // Clear by re-initializing the speculative logs.
             init_speculative_logs(signature_verified_block.len());
 
-            ret = self.execute_transactions_sequential(
+            (ret, has_reconfiguration) = self.execute_transactions_sequential(
                 executor_arguments,
                 &signature_verified_block,
                 base_view,
             )
         }
 
+        if let Ok(ref mut outputs) = ret {
+            if !has_reconfiguration {
+                if let Some(txn) = state_checkpoint_txn {
+                    // If the block execution is halted due to reconfiguration,
+                    // StateCheckpoint is not added for back-forward compatibility.
+                    // Otherwise, add back the output of the StateCheckpoint txn.
+                    let executor = E::init(executor_arguments);
+                    let res = executor.execute_transaction(
+                        &LatestView::<T, S>::new_btree_view(base_view, &BTreeMap::new(), 0),
```

Comment:
This will not observe the changes from the block itself, right? If we wanted to do that we would have to use the view that was used during the block execution.